src/modules/module_29600.c

```diff
@@ -143,6 +144,13 @@ int module_hash_decode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSE
   terra->salt_buf[2] = hex_to_u32 (salt_pos + 16);
   terra->salt_buf[3] = hex_to_u32 (salt_pos + 24);
 
+  // store IV
+  if (iv_len != 32) return (PARSER_SALT_LENGTH);
+  terra->iv[0] = hex_to_u32 (iv_pos +  0);
+  terra->iv[1] = hex_to_u32 (iv_pos +  8);
+  terra->iv[2] = hex_to_u32 (iv_pos + 16);
+  terra->iv[3] = hex_to_u32 (iv_pos + 24);
```

Comment:
Cannot we use `hex_decode` here?

```suggestion
  hex_decode (iv_pos, iv_len, (u8 *) terra->iv);
```

Reply:
Yes, but with the exception that if the length is both fixed and limited in size we prefer hex_to_u32() because it makes it easier to read.

Reply:
Noted!